### PR TITLE
chore: revert api version to `v0`

### LINKED ISSUE #87: Revert API REST endpoints to `/v0`
### Description

Revert all endpoint prefixes in `agent_api_rest` from `/v1` to `/v0`.

### Motivation

The motivation for this is that we want to indicate that the API is still in a pre-release stage. This means the API is under active development, and breaking changes may occur before it reaches a stable version 1.0.

### Resources

n/a

### To-do List

- [ ] revert to `/v0`

### PR DESCRIPTION
**Pull request by nanderstabel**
# Description of change
Revert all endpoint prefixes in agent_api_rest from /v1 to /v0.

(see #87 for the motivation)

## Links to any relevant issues
fixes #87 

## How the change has been tested
n/a

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have successfully tested this change in a docker environment

### CODE REVIEW

File: agent_api_rest/src/lib.rs [hunk 1]
```diff
@@ -63,9 +68,14 @@ pub fn app(state: ApplicationState) -> Router {
         .route(&path("/auth/token"), post(token))
         .route(&path("/openid4vci/credential"), post(credential))
         // Agent Verification Preparations
-        .route(&path("/v1/authorization_requests"), post(authorization_requests))
         .route(
-            &path("/v1/authorization_requests/:authorization_request_id"),
+            &path(&format!("/{API_VERSION}/authorization_requests")),
```
Comment (Norlock) [inline agent_api_rest/src/lib.rs]: Maybe you can fix this directly in the path function it will make it a lot more compact.

**Review by Norlock — CHANGES_REQUESTED** [review]

File: agent_api_rest/src/lib.rs (hunk 1 above)
Comment (nanderstabel) [reply agent_api_rest/src/lib.rs]: hmm the problem here though is that not all endpoints have  the API version prefix :thinking:

Comment (nanderstabel) [reply agent_api_rest/src/lib.rs]: but I can solve it by using this probably: https://docs.rs/axum/latest/axum/struct.Router.html#method.nest

I'll give that a go :+1:

Comment (Norlock) [reply agent_api_rest/src/lib.rs]: Yes you can use that or maybe have a `path_ver` fn parallel to it.

Comment (nanderstabel) [reply agent_api_rest/src/lib.rs]: Ok I've changed it to this now:
```rust
    pub const API_VERSION: &str = "/v0";

    ...

    Router::new()
        .nest(
            &path(API_VERSION),
            Router::new()
                // Agent Issuance Preparations
                .route(
                    "/configurations/credential_configurations",
                    post(configurations::credential_configurations),
                )
                .route("/credentials", post(credentials))
                .route("/credentials/:credential_id", get(get_credentials))
                .route("/offers", post(offers))
                // Agent Verification Preparations
                .route("/authorization_requests", post(authorization_requests))
                .route(
                    "/authorization_requests/:authorization_request_id",
                    get(get_authorization_requests),
                ),
        )
```

which looks cleaner I think

File: agent_application/docker/docker-compose.yml [hunk 2]
```diff
@@ -38,6 +38,7 @@ services:
       AGENT_CONFIG_EVENT_STORE: postgres
       # AGENT_CONFIG_BASE_PATH: "unicore"
       AGENT_APPLICATION_URL: ${AGENT_APPLICATION_URL}
+      AGENT_CONFIG_URL: ${AGENT_APPLICATION_URL}
```
Comment (Norlock) [inline agent_application/docker/docker-compose.yml]: Moet deze niet AGENT_CONFIG_URL zijn? Aangezien ze nu beide AGENT_APPLICATION_URL zijn.

Comment (nanderstabel) [reply agent_application/docker/docker-compose.yml]: ja beter om dat ook gelijk te doen idd :+1:

**Review by Norlock — APPROVED** [review]

**Pull request merged**